test/buffers_utest.cpp

```diff
@@ -156,6 +156,8 @@ class BuffersUtest {
         assert(ok);
 
         delete vec1;
+        delete i_even;
```

Comment:
If you can wrap `vec1`, `i_even` and `i_odd` in `unique_ptr` then you don't need the explicit delete (your code still leak if an exception is raised between the allocation and the delete, with a `unique_ptr` you won't have this problem.